app/pub_sub/subscribers/site_activity.rb

```diff
@@ -0,0 +1,33 @@
+module Subscribers
+  class SiteActivity < ::Subscribers::Base
+    def site_created(event)
+      return if event.payload[:subject].nil? || event.payload[:subject].new_record?
+
+      create_activity_from_event(event)
+    end
+
+    def site_updated(event)
+      return if event.payload[:changes].empty? || !event.payload[:subject].valid?
+
+      create_activity_from_event(event)
+    end
+
+    def site_deleted(event)
+      return if !event.payload[:subject].destroyed?
+
+      create_activity_from_event(event)
+    end
+
+    private
+
+    def create_activity_from_event(event)
```

Comment:
I'd say storing record changes would be great for providing context. How about keeping them in raw format, whenever available?